common/djangoapps/terrain/stubs/http.py

```diff
@@ -217,6 +218,8 @@ def _format_msg(self, format_str, *args):
         `format_str` is a string with old-style Python format escaping;
         `args` is an array of values to fill into the string.
         """
+        if not args:
+            format_str = urllib.unquote(format_str)
```

Comment:
Thanks for the explanation of the method @muzaffaryousaf.  

I'm trying to learn more about our practices and figure out whether:
1. This method is really `_format_url_encoded_msg()`, or we feel that non-url-encoded strings would be safe to unquote so it doesn't matter.
2. Do we typically pass course id around the platform url-encoded as a matter of practice?
@nedbat Since I'm working with you on best practices, I'm curious if you could shed any light on this.  Thanks!